Bottom: 1e12c73c4a36ccb38d0e52a39292e55557f3f8c9
Top:    9afbfd94dd518275c88f76d91d238ea7745ccfaa
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-08-07 14:52:34 +0800

Refresh of net-core-introduce-xdp-rx

---

diff --git a/include/linux/netdevice.h b/include/linux/netdevice.h
index 282e2e9..f305d4d 100644
--- a/include/linux/netdevice.h
+++ b/include/linux/netdevice.h
@@ -420,6 +420,7 @@ enum rx_handler_result {
 };
 typedef enum rx_handler_result rx_handler_result_t;
 typedef rx_handler_result_t rx_handler_func_t(struct sk_buff **pskb);
+typedef rx_handler_result_t rx_handler_xdp_func_t(struct xdp_buff *xdp);
 
 void __napi_schedule(struct napi_struct *n);
 void __napi_schedule_irqoff(struct napi_struct *n);
@@ -1898,6 +1899,7 @@ struct net_device {
 	struct bpf_prog __rcu	*xdp_prog;
 	unsigned long		gro_flush_timeout;
 	rx_handler_func_t __rcu	*rx_handler;
+	rx_handler_func_xdp_t __rcu *rx_handler_xdp;
 	void __rcu		*rx_handler_data;
 
 #ifdef CONFIG_NET_CLS_ACT
